mir-ci/mir_ci/robot_libraries/Screencopy.py

```diff
@@ -0,0 +1,137 @@
+import asyncio
+import base64
+import os
+import time
+from io import BytesIO
+from typing import List
+
+from mir_ci.screencopy_tracker import ScreencopyTracker
+from PIL import Image
+from robot.api import logger
+from robot.api.deco import keyword, library
+from RPA.Images import Images
+from RPA.recognition.templates import ImageNotFoundError
+
+
+@library(scope="GLOBAL")
+class Screencopy(ScreencopyTracker):
+    """
+    A Robot Framework library for capturing screenshots from the
+    Wayland display and performing image template matching.
+
+    The client connects to the display upon entering the first keyword,
+    and disconnects when the library goes out of scope.
+
+    If WAYLAND_DISPLAY is not defined, it defaults to 'wayland-0'.
+    """
+
+    ROBOT_LISTENER_API_VERSION = 3
+    TOLERANCE = 0.8
+
+    def __init__(self) -> None:
+        self.ROBOT_LIBRARY_LISTENER = self
+        display_name = os.environ.get("WAYLAND_DISPLAY", "wayland-0")
+        super().__init__(display_name)
+        self._rpa_images = Images()
+        self.screenshot = None
+
+    @keyword
+    async def match(self, template: str, timeout: int = 5) -> List[dict]:
+        """
+        Grab screenshots and compare until there's a match with the provided
+        template.
+
+        :param template: path to an image file to be used as template
+        :param timeout: timeout in seconds
+        :return: list of matched regions
+        :raises ImageNotFoundError: if no match is found within the timeout
+        """
+        regions = []
+        end_time = time.time() + float(timeout)
+        last_checked_frame_count = 0
+        while time.time() <= end_time:
+            self.screenshot = await self.grab_screenshot()
```

Comment:
We need to time out here:

```suggestion
            self.screenshot = await asyncio.wait_for(self.grab_screenshot(), timeout)
```

Reply:
Done.